### ISSUE TITLE
Support MTS TypeScript files

### ISSUE THREAD

**Issue by rorymalcolm** [github/linguist#5696]
## Describe the enhancement

As of [TypeScript 4.5](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5-beta/), TypeScript supports `.mts` and `.mcs` files as possible extensions. Linguist should recognise these files as TypeScript files and highlight / index them accordingly.

**Comment by lildude** [comment]
I suspect you didn't plan on closing this so I'm reopening. I also think you mean `.cts` and not `.mcs`.

Please note, support won't be added, ie any PR adding support won't be merged, until there is widespread use of the extensions (see CONTRIBUTING.md).

**Comment by rorymalcolm** [comment]
> I suspect you didn't plan on closing this so I'm reopening. I also think you mean `.cts` and not `.mcs`.
> 
> 
> 
> Please note, support won't be added, ie any PR adding support won't be merged, until there is widespread use of the extensions (see CONTRIBUTING.md).

Hi, I did mean to close, although it's in wide use on GitHub (mostly within node_modules folders), official support has been delayed and therefore I didn't want to commit to it.

**Comment by Nixinova** [comment]
MTS is used in 2K+ repos even when I restrict search to just node_modules/*/index.d.mts, so that should be alright  at least. [(link)](https://github.com/search?p=1&q=extension%3Amts+path%3Anode_modules+filename%3Aindex.d.mts&type=Code)

**Comment by rorymalcolm** [comment]
> MTS is used in 2K+ repos even when I restrict search to just node_modules/*/index.d.mts, so that should be alright at least. [(link)](https://github.com/search?p=1&q=extension%3Amts+path%3Anode_modules+filename%3Aindex.d.mts&type=Code)

Yeah althought it has widespread usage and semi support it's still the case that its not _official_ yet, was meant to be in TS 4.6 but has since potentially slipped, see: https://github.com/microsoft/TypeScript/issues/46452 for further info

**Comment by Inkdpixels** [comment]
@rorymalcolm Not sure if it really slipped, the issue you are referencing clearly states that this is the standard that they are aiming to establish, no?

Also the `.mts` / `. cts` extensions are announced in the 4.7 beta blog post

https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#new-file-extensions

**Comment by rorymalcolm** [comment]
> @rorymalcolm Not sure if it really slipped, the issue you are referencing states that this is the standard that they are aiming to establish, no?
> 
> Also the `.mts` / `. cts` extensions are announced in the 4.7 beta blog post
> 
> https://devblogs.microsoft.com/typescript/announcing-typescript-4-7-beta/#new-file-extensions

Hadn't seen their inclusion 4.7's beta, probably fine, I just was aware there was some debate over it in the community when I wrote that comment months ago!
